### DISCUSSION TITLE
Mouse dragging lag and solution

### DISCUSSION THREAD

**Discussion by EYHN** [EYHN/vscode-vibrancy#80] [General]
## Problem
In Windows 10 higher than version 1809, after enabling this extension, dragging the window with the mouse will cause a very serious lag.

After my research found, it seems to be a problem inside DWM, the mouse polling rate is higher than the screen rate, causing the rendering request to block in the queue, and it looks like a mouse delay.

## current solution 
Learning from [Seo-Rii's solution in his project](https://github.com/Seo-Rii/electron-acrylic-window/issues/6), I've implemented a window move/resize throttle, it can skip some move/resize events to match the screen refresh rate.

This solution was added to this extension from `v1.0.10`. In my test, it performs well on the 144hz high refresh rate display, and there is no mouse lag problem. On a 60hz display, there is still a little lag, but it is acceptable.

This is not a perfect solution at now, we can discuss it here. 

## Related issues
https://github.com/EYHN/vscode-vibrancy/issues/5 https://github.com/23phy/ewc/issues/22 https://github.com/reycn/listen1_desktop_fluent/issues/15 https://github.com/EYHN/vscode-vibrancy/issues/73

**Comment by lzlrd** [comment]
For what its worth, https://github.com/Seo-Rii/electron-acrylic-window/issues/26#issuecomment-715902985 suggests enabling Vibrancy's `useCustomWindowRefreshMethod` toggle to avoid movement/resize lag.

**Reply by drementer** [reply]
> For what its worth, [Seo-Rii/electron-acrylic-window#26 (comment)](https://github.com/Seo-Rii/electron-acrylic-window/issues/26#issuecomment-715902985) suggests enabling Vibrancy's `useCustomWindowRefreshMethod` toggle to avoid movement/resize lag.

I am very confused please can you tell me how to solve the problem step by step?

**Reply by drementer** [reply]
First of all, thank you for answering. But unfortunately, even though I tried this before, there is no change. And yes, I just tried again. And also I deleted all custom settings in my settings file but similarly didn't affect me. 
Not: As a solution to the problem, I ran the Blur Test.exe file on the page they directed above and it was working fine.

**Reply by lzlrd** [reply]
@drementer, I haven't looked into it much but `useCustomWindowRefreshMethod` is part of a electron-acrylic-window wrapper around Electron's vibrancy (which this extension uses directly). You'd have to modify this extension to use the wrapper, and then modify the `setVibrancy()` call as per https://github.com/Seo-Rii/electron-acrylic-window#options for Windows clients.

**Comment by cykvta** [comment]
On 75hrz display and win11 21H2 still happens.

**Comment by leonasdev** [comment]
On 240hz display and Win10 21H2 still happens.
